native/cocos/base/Assertf.h

```diff
@@ -0,0 +1,172 @@
+    #endif
+
+    // NOLINTNEXTLINE
+    #define _CC_ASSERT_(cond) assert(cond)
+
+    // NOLINTNEXTLINE
+    #define _CC_ASSERTF_CMP(expr1, op, expr2, fmt, ...)             \
+        do {                                                        \
+            CC_ASSERT_FORMAT((expr1)op(expr2), fmt, ##__VA_ARGS__); \
+        } while (false)
+
+    // NOLINTNEXTLINE
+    #define _CC_ASSERT_CMP(expr1, op, expr2) \
+        do {                                 \
+            _CC_ASSERT_((expr1)op(expr2));   \
+        } while (false)
+
+    // CC_ASSERTF variants
+    #define CC_ASSERTF_EQ(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, ==, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_NE(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, !=, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_LE(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, <=, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_GE(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, >=, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_LT(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, <, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_GT(expr1, expr2, fmt, ...) _CC_ASSERTF_CMP(expr1, >, expr2, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_TRUE(expr1, fmt, ...)      _CC_ASSERTF_CMP(expr1, ==, true, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_FALSE(expr1, fmt, ...)     _CC_ASSERTF_CMP(expr1, ==, false, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_NULL(expr1, fmt, ...)      _CC_ASSERTF_CMP(expr1, ==, nullptr, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_NOT_NULL(expr1, fmt, ...)  _CC_ASSERTF_CMP(expr1, !=, nullptr, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_ZERO(expr1, fmt, ...)      _CC_ASSERTF_CMP(expr1, ==, 0, fmt, ##__VA_ARGS__)
+    #define CC_ASSERTF_NONZERO(expr1, fmt, ...)   _CC_ASSERTF_CMP(expr1, !=, 0, fmt, ##__VA_ARGS__)
+    // CC_ASSERT variants
+    #define CC_ASSERT_EQ(expr1, expr2) _CC_ASSERT_CMP(expr1, ==, expr2)
+    #define CC_ASSERT_NE(expr1, expr2) _CC_ASSERT_CMP(expr1, !=, expr2)
+    #define CC_ASSERT_LE(expr1, expr2) _CC_ASSERT_CMP(expr1, <=, expr2)
+    #define CC_ASSERT_GE(expr1, expr2) _CC_ASSERT_CMP(expr1, >=, expr2)
+    #define CC_ASSERT_LT(expr1, expr2) _CC_ASSERT_CMP(expr1, <, expr2)
+    #define CC_ASSERT_GT(expr1, expr2) _CC_ASSERT_CMP(expr1, >, expr2)
+    #define CC_ASSERT_TRUE(expr1)      _CC_ASSERT_CMP(expr1, ==, true)
+    #define CC_ASSERT_FALSE(expr1)     _CC_ASSERT_CMP(expr1, ==, false)
+    #define CC_ASSERT_NULL(expr1)      _CC_ASSERT_CMP(expr1, ==, nullptr)
+    #define CC_ASSERT_NOT_NULL(expr1)  _CC_ASSERT_CMP(expr1, !=, nullptr)
+    #define CC_ASSERT_ZERO(expr1)      _CC_ASSERT_CMP(expr1, ==, 0)
+    #define CC_ASSERT_NONZERO(expr1)   _CC_ASSERT_CMP(expr1, !=, 0)
+    /**
+     * @brief printf like assert
+     *
+     *  CC_ASSERTF(1==2, "value %d should not be equal to %d", 1, 2);
+     *  CC_ASSERTF_EQ(1, 3/3, "n equals to n");
+     *  CC_ASSERTF_NE(1, s, "not initial value");
+     *
+     */
```

Comment:
Please also add comment for `CC_ABORT()`

Reply:
fixed